crates/re_arrow_store/src/store_gc.rs

```diff
@@ -222,77 +233,213 @@ impl DataStore {
         // 2. Find all tables that potentially hold data associated with that `RowId`
         // 3. Drop the associated row and account for the space we got back
 
-        for (&row_id, (timepoint, entity_path_hash)) in &self.metadata_registry.registry {
+        let batch_size = (self.config.indexed_bucket_num_rows as usize).saturating_mul(2);
```

Comment:
I think I puzzled myself together by now how we accumulate these batches and then drop them in bulk, but a few guiding comments on the steps involved would have been useful :)

Reply:
Oh yes, as I ended up disabling it by default I completely forgot to explain what's going on :no_mouth: